tests/test-dirs/destruct/destruct-fun.t

```diff
@@ -102,3 +102,55 @@ FIXME UPGRADE 5.2: this was working before the upgrade
     ],
     "notifications": []
   }
+
+  $ cat >fun.ml <<EOF
+  > type t = { foo: int }
+  > let f a (b: t) c = something
+  > EOF
+
+  $ $MERLIN single case-analysis -start 2:10 -end 2:10 \
+  > -filename fun.ml <fun.ml | \
+  > sed -e 's/, /,/g' | sed -e 's/ *| */|/g' | tr -d '\n' | jq '.'
+  {
+    "class": "return",
+    "value": [
+      {
+        "start": {
+          "line": 2,
+          "col": 9
+        },
+        "end": {
+          "line": 2,
+          "col": 10
+        }
+      },
+      "{ foo }"
+    ],
+    "notifications": []
+  }
+
+  $ cat >fun.ml <<EOF
+  > type t = Foo
+  > let f a (b: t) c = something
+  > EOF
+
+  $ $MERLIN single case-analysis -start 2:10 -end 2:10 \
+  > -filename fun.ml <fun.ml | \
+  > sed -e 's/, /,/g' | sed -e 's/ *| */|/g' | tr -d '\n' | jq '.'
+  {
+    "class": "return",
+    "value": [
+      {
+        "start": {
+          "line": 2,
+          "col": 9
+        },
+        "end": {
+          "line": 2,
+          "col": 10
+        }
+      },
+      "Foo"
+    ],
+    "notifications": []
+  }
```

Comment:
It would also be good to have tests with:

* A type with one constructor where the constructor has fields.
* A type with one constructor where the constructor has an inline record
* A gadt with multiple constructors, but where the types allow only one constructor to be well-typed. (This is harder -- a reach goal.)

Reply:
Hi @goldfirere 
- Works fine
- As the previous version, it generate `Ctor _`, but currently, the way that pattern are printed seems wrong. For `type t = Foo of {..}` it produces `Foo_`. I am currently investigating on that
- With GADT it seems working too